### PR TITLE
cmd/snap: add 'snap grant'

### PR DESCRIPTION
**Pull request by zyga**
This branch adds the command for granting skills. It depends on 277c970 which is proposed separately in #449.

### CODE REVIEW

**Comment by chipaca** [comment]
:+1:

File: cmd/snap/cmd_grant.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+package main
+
+import (
+	"github.com/ubuntu-core/snappy/i18n"
+)
+
+type cmdGrant struct {
+	Positionals struct {
+		Offer SnapAndName `positional-arg-name:"<snap>:<skill>" description:"snap offering the skill" skip-help:"true" required:"true"`
```
Comment (chipaca) [inline cmd/snap/cmd_grant.go]: i'm guessing skip-help is going away

Comment (zyga) [inline cmd/snap/cmd_grant.go]: Yep, I'm about to refresh this.

File: cmd/snap/cmd_grant.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	})
+}
+
+func (x *cmdGrant) Execute(args []string) error {
+	// snap grant <skill> <snap>[:<skill slot>]
+	if x.Positionals.Offer.Snap != "" && x.Positionals.Offer.Name == "" {
+		// Move the value of .Snap to .Name and keep .Snap empty
+		x.Positionals.Offer.Name = x.Positionals.Offer.Snap
+		x.Positionals.Offer.Snap = ""
+	}
```
Comment (pedronis) [inline cmd/snap/cmd_grant.go]: this feels a bit of a hack given that we went all the way to have types for Use and Offer, seems we would need SnapSkill and SnapSlot types for Offer and Use respectively

Comment (zyga) [inline cmd/snap/cmd_grant.go]: Hmm

I'd have to think, please look at revoke in #458 

The thing I'm wondering what you said is true given the snap revoke and snap skills and their custom parsing "logic". I'm happy to do it, just need to see how the parser behavies in all the cases at once.

**Pull request merged**